test_provider.py

```diff
@@ -0,0 +1,454 @@
+                          timedelta(milliseconds=json[self._ObsKeys.SETUPTIME.value]),
+                          None,
+                          None,
+                          None,
+                          self.obs_classes[json[self._ObsKeys.OBS_CLASS.value]],
+                          targets,
+                          guide_stars,
+                          atoms,
+                          constraints,
+                          None)
+        obs.exec_time = sum([atom.exec_time for atom in atoms], timedelta()) + obs.acq_overhead
+
+        return obs
+    
+    def parse_time_allocation(self, json: dict) -> TimeAllocation:
+        return TimeAllocation(TimeAccountingCode(json[self._TAKeys.CATEGORIES.value][0][self._TAKeys.CATEGORY.value]),
+                              timedelta(milliseconds=json[self._TAKeys.AWARDED_TIME.value]),
+                              timedelta(milliseconds=0),
+                              timedelta(milliseconds=json[self._TAKeys.CATEGORIES.value][0][self._TAKeys.PROGRAM_TIME.value]),
+                              timedelta(milliseconds=json[self._TAKeys.CATEGORIES.value][0][self._TAKeys.PARTNER_TIME.value]))
+
+    def parse_or_group(self, json: dict) -> OrGroup:
+        
+        # Find nested OR groups/AND groups
+        # TODO: is this correct if there are not nested groups in OCS natively
+
+        observations = [self.parse_observation(json[key], key) for key in json.keys() if key.startswith('OBSERVATION_BASIC')]
+
+        number_to_observe = len(observations)
+        delay_max, delay_min = 0, 0 # TODO: What are these?
+        or_group = OrGroup(json['key'], json['name'], number_to_observe, delay_min, delay_max, observations)
+        return or_group
+
+    @staticmethod
+    def parse_guide_star(json: dict) -> Resource:
+
+        # TODO: Maybe a GuideStart class should be in place to handle this better
+        res = Resource(json['name'], json['tag'])
+        return res
+    
+    def parse_root_group(self, json: dict) -> AndGroup:
+        # Find nested OR groups/AND groups
+        groups = [self.parse_and_group(json[key]) for key in json.keys() if key.startswith(self._GroupsKeys.KEY.value)]
+        if any(key.startswith(self._GroupsKeys.ORGANIZATIONAL_FOLDER.value) for key in json.keys()):
+            for key in json.keys():
+                if key.startswith(self._GroupsKeys.ORGANIZATIONAL_FOLDER.value):
+                    groups.append(self.parse_or_group(json[key]))
+        num_to_observe = len(groups)
+        root_group = AndGroup(None, None, num_to_observe, 0, 0, groups, AndOption.ANYORDER)
+        return root_group
+
+    def parse_and_group(self, json: dict) -> AndGroup:
+        observations = [self.parse_observation(json[key], key) for key in json.keys() if key.startswith(self._ObsKeys.KEY.value)]
+
+        number_to_observe = len(observations)
+        delay_max, delay_min = 0, 0 # TODO: What are these?
```

Comment:
These have to do with cadences, I believe. Bryan goes into some detail on them in his document. I don't think we have to worry about them too much in the initial implementation, but if the information is available, we should populate it. They will likely become more important as we connect things to GPP.